### PR TITLE
Classes & Props: Checkbox

### PR DESCRIPTION
**Pull request by levithomason**

### CODE REVIEW

File: src/components/Checkbox/Checkbox.js [hunk 1]
```diff
@@ -19,8 +19,9 @@ import $ from 'jquery';
  * <Checkbox type='toggle' />
  */
 @radium
-class Checkbox extends Component {
+export default class Checkbox extends Component {
```
Comment (levithomason) [inline src/components/Checkbox/Checkbox.js:23]: Export inline, instead of at the end of the file.

File: src/components/Checkbox/Checkbox.js [hunk 2]
```diff
@@ -29,6 +30,11 @@ class Checkbox extends Component {
     type: PropTypes.string,
   };
 
+  static defaultProps = {
+    name: 'checkbox',
+    type: 'checkbox',
+  };
```
Comment (levithomason) [inline src/components/Checkbox/Checkbox.js:37]: Removed `||` in favor of defaultProps.

File: src/components/Checkbox/Checkbox.js [hunk 3]
```diff
@@ -45,16 +51,18 @@ class Checkbox extends Component {
...
-    let name = this.props.name || 'checkbox';
-    let hasLabel = !!this.props.label;
-    let classes = classNames('sd-checkbox', 'ui', {fitted: !hasLabel}, this.props.type, 'checkbox');
+    let classes = classNames(
+      'sd-checkbox',
+      'ui',
+      this.props.className,
+      {fitted: !this.props.label},
+      'checkbox'
+    );
```
Comment (levithomason) [inline src/components/Checkbox/Checkbox.js:61]: Inherit className.

File: src/components/Checkbox/Checkbox.js [hunk 4]
```diff
@@ -45,16 +51,18 @@ class Checkbox extends Component {
...
     return (
-      <div className={classes} style={this.props.style} ref='container'>
-        <input defaultChecked={this.props.defaultChecked} type='checkbox' name={name} ref='checkbox' />
+      <div className={classes} ref='container'>
+        <input {...this.props} ref='checkbox' />
```
Comment (levithomason) [inline src/components/Checkbox/Checkbox.js]: Spread props.

File: test/src/components/Checkbox-test.js [hunk 5]
```diff
@@ -2,17 +2,18 @@ import React from 'react';
...
   it('should have a semantic ui plugin to handle the check action', () => {
-    let renderedCheckbox = render(<Checkbox name='firstName' label='Include First' />).first();
-    renderedCheckbox.container.checkbox.called.should.equal(true);
+    render(<Checkbox name='firstName' label='Include First' />)
+      .first()
+      .container.checkbox.called.should.equal(true);
   });
   it('should have a fitted class if no label is given', () => {
-    render(<Checkbox name='firstName' />).findClass('sd-checkbox').props.className.should.contain('fitted');
+    render(<Checkbox name='firstName' />).findClass('fitted');
```
Comment (levithomason) [inline test/src/components/Checkbox-test.js:17]: Updated test syntax whilst debugging failing tests.  No functional changes, just syntax here.

File: src/components/Checkbox/Checkbox.js [hunk 6]
```diff
@@ -45,16 +52,20 @@ class Checkbox extends Component {
...
+      {fitted: !this.props.label},
+      'checkbox'
+    );
+    let checkboxProps = _.clone(this.props);
+    delete checkboxProps.className;
     return (
-      <div className={classes} style={this.props.style} ref='container'>
-        <input defaultChecked={this.props.defaultChecked} type='checkbox' name={name} ref='checkbox' />
+      <div className={classes} ref='container'>
+        <input {...checkboxProps} ref='checkbox' />
```
Comment (levithomason) [inline src/components/Checkbox/Checkbox.js:66]: Only spread non class props, since classes belong on the containing div.

Comment (athurman) [inline src/components/Checkbox/Checkbox.js:66]: nifty, will we plan to implement this with every stardust component?

**Comment by athurman** [comment]
Looks good! :ocean: :octopus:

File: src/components/Checkbox/Checkbox.js (hunk 6 above)
Comment (eanplatter) [inline src/components/Checkbox/Checkbox.js:66]: haha this way cleaner than trying to determine whether or not it's a className >.<  +1

**Comment by eanplatter** [comment]
:fallen_leaf:

File: src/components/Checkbox/Checkbox.js (hunk 6 above)
Comment (levithomason) [inline src/components/Checkbox/Checkbox.js:66]: @athurman yep, every component is getting this update.

**Pull request merged**